jerry-core/api/jerry.c

```diff
@@ -447,6 +447,26 @@ jerry_eval (const jerry_char_t *source_p, /**< source code */
 } /* jerry_eval */
 
 /**
+ * Run enqueued Promise jobs until the first thrown error or until all get executed.
+ *
+ * Note:
+ *      returned value must be freed with jerry_release_value, when it is no longer needed.
+ *
+ * @return result of last executed job, may be error value.
+ */
+jerry_value_t
+jerry_run_all_enqueued_jobs (void)
```

Comment:
I would prefer a `jerry_run_enqueued_jobs` with some parameter which provides some control. The problem is that a job can create other jobs, and this could lead to an infinite loop, which might not be desired by the platform. So it would be good to have a termination condition (e.g. max number of executed jobs, -1 for any).